File: poc-kronos-frontend/src/model/basecurricular/Habilidade.js
class Habilidade {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.idNivel = dados ? dados.idNivel : null;
    this.nivel = dados ? dados.nivel : null;
    this.idComponente = dados ? dados.idComponente : null;
    this.componente = dados ? dados.componente : null;
    this.descricao = dados ? dados.descricao : null;
    this.codigo = dados ? dados.codigo : null;
    this.ativo = dados ? dados.ativo : true;
    this.bncc = dados ? dados.bncc : null;
    this.eixo = dados ? dados.eixo : null;
    this.campoAtuacao = dados ? dados.campoAtuacao : null;
    this.praticaLinguagem = dados ? dados.praticaLinguagem : null;
    this.objetoConhecimento = dados ? dados.objetoConhecimento : null;
    this.unidadeTematica = dados ? dados.unidadeTematica : null;
    this.idsCompetencias = dados ? dados.idsCompetencias : [];
    this.idsFaixasAnos = dados ? dados.idsFaixasAnos : null;
  }

  static validations(required, maxLength) {
    return {
      descricao: { required, maxLength: maxLength(1500) },
      codigo: { required, maxLength: maxLength(5) },
      idNivel: { required },
      // idsCompetencias: { required },
      idsFaixasAnos: { required },
      eixo: { maxLength: maxLength(100) },
      campoAtuacao: { maxLength: maxLength(300) },
      praticaLinguagem: { maxLength: maxLength(300) },
      objetoConhecimento: { maxLength: maxLength(500) },
      unidadeTematica: { maxLength: maxLength(300) }
    };
  }

  static validationMessages() {
    return {
      descricao: {
        required: "Descrição é obrigatória",
        maxLength: "Descrição só aceita no máximo 500 caracteres"
      },
      codigo: {
        required: "Código é obrigatório",
        maxLength: "Código só aceita no máximo 5 caracteres"
      },
      idNivel: {
        required: "Nível é obrigatório"
      },
      // idsCompetencias: {
      //   required: "Competência é obrigatória"
      // },
      idsFaixasAnos: {
        required: "Faixa/Ano é obrigatório"
      }
    };
  }
}
export default Habilidade;

File: src/main/java/br/com/kronos/backend/aplicacao/basecurricular/Habilidade.java


package br.com.kronos.backend.aplicacao.basecurricular;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Habilidade {
    
    private long id;
    private Long idComponente;
    private String descricao;
    private String codigo;
    private boolean ativo;
    private Boolean bncc;
    private String eixo;
    private String campoAtuacao;
    private String praticaLinguagem;
    private String objetoConhecimento;
    private String unidadeTematica;
    private Long idInstituicao;

	@Builder
    public Habilidade(long id, Long idComponente, String descricao, String codigo, boolean ativo, Boolean bncc, String eixo, 
          String campoAtuacao, String praticaLinguagem, String objetoConhecimento, String unidadeTematica, Long idInstituicao) {

		super();
        this.id = id;
        this.idComponente = exigirNaoNulo(idComponente, "Componente"); 
        this.descricao = exigirTamanho(descricao, "Descrição", 1, 1500); 
        this.codigo = exigirTamanho(codigo, "Código", 1, 10); 
        this.ativo = exigirNaoNulo(ativo, "Ativo");
        this.eixo = exigirTamanho(eixo, "Eixo", 0, 100);  
        this.campoAtuacao = exigirTamanho(campoAtuacao, "Campo Atuação", 0, 300);  
        this.praticaLinguagem = exigirTamanho(praticaLinguagem, "Prática da Linguagem", 0, 300);   
        this.objetoConhecimento = exigirTamanho(objetoConhecimento, "Objeto Conhecimento", 0, 500); 
        this.unidadeTematica = exigirTamanho(unidadeTematica, "Unidade Temática", 0, 300);
        if (bncc != null && bncc) {
        	this.bncc = bncc;
        	this.idInstituicao = null; 
        } else {
        	this.bncc = false;
        	this.idInstituicao = idInstituicao;     
        }    
	}
}